src/cli/eofwrap.py

```diff
@@ -0,0 +1,370 @@
+
+        out_fixtures = BaseFixturesRootModel({})
+        fixture: BlockchainFixture
+        for id, fixture in fixtures.items():
+            fixture_eof_codes = []
+            wrapped_at_least_one_account = False
+
+            if fixture.pre:
+                for address, account in fixture.pre.root.items():
+                    if account is None or account.code is None or len(account.code) == 0:
+                        continue
+                    wrapped = wrap_code(account.code)
+                    if self._validate_eof(wrapped):
+                        account.code = Bytes(wrapped)
+                        wrapped_at_least_one_account = True
+                        self.metrics[self.ACCOUNTS_WRAPPED] += 1
+                        fixture_eof_codes.append(to_hex(account.code))
+
+                        # wrap the same account in post state the same way
+                        if fixture.post_state and fixture.post_state.root[address]:
+                            fixture.post_state.root[address].code = Bytes(wrapped)  # type: ignore
+                    else:
+                        self.metrics[self.ACCOUNTS_INVALID_EOF] += 1
+            if not wrapped_at_least_one_account:
+                self.metrics[self.FIXTURES_CANT_WRAP] += 1
+                continue
+
+            try:
+                out_fixture = self._wrap_fixture(fixture, traces)
+                out_fixtures[id] = out_fixture
+                self.metrics[self.FIXTURES_GENERATED] += 1
+                self.unique_eof.update(fixture_eof_codes)
+                self.metrics[self.UNIQUE_ACCOUNTS_WRAPPED] = len(self.unique_eof)
+            except Exception as e:
+                short = str(e)
+                if len(short) > GENERATION_ERROR_MESSAGE_THRESHOLD:
+                    short = short[:GENERATION_ERROR_MESSAGE_THRESHOLD] + "..."
+
+                _inc_counter(self.metrics[self.GENERATION_ERRORS], short)
+
+                self.metrics[self.FIXTURES_CANT_GENERATE] += 1
+                self.metrics[self.ACCOUNTS_CANT_GENERATE] += len(fixture_eof_codes)
+
+        if len(out_fixtures) == 0:
+            self.metrics[self.FILES_SKIPPED] += 1
+            return
+
+        os.makedirs(os.path.dirname(out_path), exist_ok=True)
+        out_fixtures.collect_into_file(Path(out_path))
+        self.metrics[self.FILES_GENERATED] += 1
+
+    def _wrap_fixture(self, fixture: BlockchainFixture, traces: bool):
+        env = Environment()
+
+        pre = fixture.pre
+
+        t8n = EvmOneTransitionTool(trace=traces)
+
+        test = BlockchainTest(
+            genesis_environment=env,
+            pre=pre.root,
+            post=fixture.post_state.root if fixture.post_state else {},
+            blocks=[],
+            tag="wrapped test",
+        )
+
+        for fixture_block in fixture.blocks:
+            if isinstance(fixture_block, FixtureBlock):
+                header = fixture_block.header
+                block = Block(
```

Comment:
I think this should focus on the transactions part of the fixture, because, for example, overriding the state root with the one in the original fixture is almost always going to generate an invalid block.